ocaml/xapi/livepatch.ml

```diff
@@ -82,7 +82,7 @@ let get_latest_livepatch lps =
 module BuildId = struct
   let one_byte =
     let open Angstrom in
-    any_uint8 >>= fun c -> return (Printf.sprintf "%0x" c)
+    any_uint8 >>= fun c -> return (Printf.sprintf "%02x" c)
```

Comment:
This code reads just one byte, so `%02x` is correct. The build ID by itself might be larger and there would be better functions in Angstrom to read 16, 32, or 64-bit values because endian-ness needs to be considered

Reply:
> The build ID by itself might be larger and there would be better functions in Angstrom to read 16, 32, or 64-bit values because endian-ness needs to be considered

Sorry, I can't see the problem here? May I please have more elaborations here?

Reply:
Is the build ID guaranteed to be 8 bit? If not, `any_uint8` is not the right function to read it. By itself, `one_byte` looks correct - it reads a byte and represents its as two-character hexadecimal number.

Reply:
> Is the build ID guaranteed to be 8 bit?
The build ID is of 20 bytes. And each each byte in it should be presented as hex with width 2.
The bug to be solved is:
correct: b5aa555f696727cc8e6d006bb78070cb95fb9e9e
wrong: b5aa555f696727cc8e6d06bb78070cb95fb9e9e
If one byte is 0, it was presented as '0', rather than '00'.

Reply:
Ah, thanks, found out what this is about.
Build id can present in the .notes section of any ELF binary, as an example:
```
eu-readelf -n /bin/bash | grep BUILD_ID -A1
  GNU                   20  GNU_BUILD_ID
    Build ID: 65289d3e4b67a5f765c63c7ec51c7f28f753ce08
```